src/compiler/checker.ts

```diff
@@ -2895,6 +2902,34 @@ namespace ts {
             return unknownType;
         }
 
+        function getTypeOfBasePropertyDeclaration(declaration: VariableLikeDeclaration) {
+            if (declaration.parent.kind === SyntaxKind.ClassDeclaration) {
+                const property = getPropertyFromBaseInterfaces(<ClassLikeDeclaration>declaration.parent, declaration.symbol.name);
+                if (property) {
+                    return getTypeOfSymbol(property);
+                }
+            }
+        }
+
+        function getPropertyFromBaseInterfaces(declaration: ClassLikeDeclaration, propertyName: string): Symbol {
+            const implementedTypeNodes = getClassImplementsHeritageClauseElements(declaration);
+            if (implementedTypeNodes) {
+                for (const typeRefNode of implementedTypeNodes) {
+                    const t = getTypeFromTypeReference(typeRefNode);
+                    if (t !== unknownType) {
+                        for (const property of getPropertiesOfType(t)) {

```

Comment:
Why the loop when you could use `getPropertyOfType` or `getPropertyOfObjectType`?